### DISCUSSION TITLE
Is there an easy way to use (parametrized) query files inside psql

### DISCUSSION THREAD

**Discussion by on3iro** [vitaly-t/pg-promise#835] [Q&A] [answered]
Hi,

it would sometimes be benefitial to run query files from psql for debugging purposes or database unit tests. However if I am not mistaken this would currently only work for non parametrized queries, because pg does not know about `${}`/`$<>`/.etc .
Is there an easy way to use regular pg-variables (via `\set`)  with pg-promise query files?

Thanks in advance :)

**Comment by vitaly-t** [comment, marked as answer]
Queries designed for use with pg-promise are quite unrelated to query files for psql. The former depend on values to be passed in from a NodeJS script. You cannot make it work via psql, apart from for queries that do not use any variables.

**Reply by on3iro** [reply]
Alright, thanks for the answer!
